### DISCUSSION TITLE
Tailwind unable to compile class which comes from the network request

### DISCUSSION THREAD

**Discussion by rutvik24** [tailwindlabs/tailwindcss#10199] [Help]
I have used a span that takes the bg-color class and ring-color class from the API. but tailwind is unable to compile it.
Here below I attached my code which shows how I used it. I think I have followed tailwind guidelines for passing the whole class instead of a single value as a variable.

`{product.color.map(item => {
              return (
                <span
                  key={item?.id}
                  className={`${item.background} ring-1 ring-offset-2 w-6 h-2 mx-3 my-2 rounded-full ${item.ring} hover:shadow-md md:h-6`} />))}` 

But this is not working can anyone suggest me any solution.

**Comment by wongjn** [comment]
Could you resupply the code? It is formatted strangely and I can't quite make head or tails of it!

**Reply by wongjn** [reply]
> ```
>               return (
>                 <span
>                   key={item?.id}
>                   className={`${item.background} ring-1 ring-offset-2 w-6 h-2 mx-3 my-2 rounded-full ${item.ring} hover:shadow-md md:h-6`} />))}```
> Here is the code
> ```
[^1]

Thank you for supplying that. Seems to work fine in a [Tailwind Play](https://play.tailwindcss.com/4lIx8rPx4C), so it seems like it may be some misconfiguration. Perhaps check:
- The `content` file globs in the Tailwind configuration sufficiently cover the file that this code resides in.
- Tailwind has compiled since this code was added to the code base.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/10199#discussioncomment-4560768

**Reply by rutvik24** [reply]
The class is coming from the the network request then it should be work or not ?

**Reply by wongjn** [reply]
Not sure which class you mean by <q>The class</q> but the class would need to be seen by Tailwind at **build** time, either in the code that is scans or via [`safelist`](https://tailwindcss.com/docs/content-configuration#safelisting-classes) for a Tailwind rule to be generated for the class.

**Comment by rutvik24** [comment]
```
{product.color.map(item => {
              return (
                <span
                  key={item?.id}
                  className={`${item.background} ring-1 ring-offset-2 w-6 h-2 mx-3 my-2 rounded-full ${item.ring} hover:shadow-md md:h-6`} />))}
```
Here is the code

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10199#discussioncomment-4560778 to keep the reply chain together.
